clients/go/admin/pkce/auth_flow_orchestrator.go

```diff
@@ -58,12 +53,38 @@ func (f TokenOrchestrator) FetchTokenFromCacheOrRefreshIt(ctx context.Context) (
 	if !token.Valid() {
 		return nil, fmt.Errorf("token from cache is invalid")
 	}
-
+	/* We need to get access tokens expiry as the Expiry on the oauth2.token is refresh Token expiry which is longer.
+	   This is inorder to solve the following scenario.
+	   1] Client requests access token and refresh token both in the same request
+	   2] Auth server issue:
+		  a) access token with expiry of 30 mins
+	      b) refresh token with expiry of 60 mins
+	   3] The combined access token has the expiry for refresh token
+	      {
+			access_token 	: "acces_token...."
+			refresh_token 	: "refresh_toke..."
+			expiry 			: <refresh_token_expiry"
```

Comment:
The spec actually specifies that expires_in should be that of the accesstoken. Have you tested with Okta? is that what you see as well? if this is what you see when you setup Google Idp then it's admin's code (in our control)...

Ref: https://tools.ietf.org/html/rfc6749#section-5.1
```
expires_in
         RECOMMENDED.  The lifetime in seconds of the access token.
```

Reply:
Opps. I was skeptical of making this change and thanks for pointing that out from RFC. This was the bug which lurking on admin side and causing issues earlier aswell but was hidden due to earlier logic of force refresh.